src/librbd/ImageCtx.cc

```diff
@@ -405,13 +405,15 @@ struct C_InvalidateCache : public Context {
     return flags;
   }
 
-  int ImageCtx::snap_set(string in_snap_name)
+  int ImageCtx::snap_set(cls::rbd::SnapshotNamespace in_snap_namespace,
+			 string in_snap_name)
   {
     assert(snap_lock.is_wlocked());
-    snap_t in_snap_id = get_snap_id(in_snap_name);
+    snap_t in_snap_id = get_snap_id(in_snap_name, in_snap_namespace);
```

Comment:
It seems like I can't change the order of namespace and name in snap_ids map. I use function: lower_bound when I look for a snapshot by its namespace when I don't know its name. Therefore namespace should go first. I'll need to change everything to be in the order: name, namespace. Unless you can change my view as you often do)